src/compiletest/runtest.rs

```diff
@@ -927,6 +927,13 @@ fn check_expected_errors(expected_errors: Vec<errors::ExpectedError>,
         format!("{}:{}:", testfile.display(), ee.line)
     }).collect::<Vec<String>>();
 
+    let (expect_help, expect_note) =
+        expected_errors.iter()
+                        .fold((false, false),
+                              |(acc_help, acc_note), ee|
+                                  (acc_help || ee.kind == "help:", acc_note ||
```

Comment:
@pnkfelix:

> In hindsight I had thought that what we were discussing was "if there is either a help or a note, then all help and notes must be present."

I actually wondered later which version we meant, and figured I'd probably be ok with either one...

@fhahn 

> By the way, is there a place to document this?

We've gone back and forth on this. I know that @brson has been working on making some kind of "wiki-like" page for compiler internal docs and the like, I'm not sure what's the current status on that? I'm not sure what happened to the existing docs on the compiletest infrastruture, but it seems like it should go in there.